tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs

```diff
@@ -310,52 +310,48 @@ private async Task UploadLogLinesBlobAsync(string account, Build build, BuildLog
                     return;
                 }
 
-                using var tempFile = new DisposableTempFile();
+                var messagesWriter = new StringBuilder();
+                var logLines = await this.logProvider.GetLogLinesAsync(build, log.Id);
+                var lastTimeStamp = log.CreatedOn;
 
-                await using (var messagesWriter = new StreamWriter(File.OpenWrite(tempFile.Path)))
+                for (var lineNumber = 1; lineNumber <= logLines.Count; lineNumber++)
                 {
-                    var logLines = await this.logProvider.GetLogLinesAsync(build, log.Id);
-                    var lastTimeStamp = log.CreatedOn;
+                    var line = logLines[lineNumber - 1];
+                    var match = Regex.Match(line, @"^([^Z]{20,28}Z) (.*)$");
+                    var timestamp = match.Success
+                        ? DateTime.ParseExact(match.Groups[1].Value, TimeFormat, null,
+                            System.Globalization.DateTimeStyles.AssumeUniversal).ToUniversalTime()
+                        : lastTimeStamp;
 
-                    for (var lineNumber = 1; lineNumber <= logLines.Count; lineNumber++)
-                    {
-                        var line = logLines[lineNumber - 1];
-                        var match = Regex.Match(line, @"^([^Z]{20,28}Z) (.*)$");
-                        var timestamp = match.Success
-                            ? DateTime.ParseExact(match.Groups[1].Value, TimeFormat, null,
-                                System.Globalization.DateTimeStyles.AssumeUniversal).ToUniversalTime()
-                            : lastTimeStamp;
+                    var message = match.Success ? match.Groups[2].Value : line;
 
-                        var message = match.Success ? match.Groups[2].Value : line;
+                    if (timestamp == null)
+                    {
+                        throw new Exception($"Error processing line {lineNumber}. No leading timestamp.");
```

Comment:
Suggestion: dump the line causing the error too for easy debugging in an unexpected case later.